### PR TITLE
Fix integration with IncrementalClassifier

### PR DESCRIPTION
**Pull request by JuliousHurtado**
When testing the RAR implementation using a model with `IncrementalClassifier` class, @AlbinSou found a new error.

The problem arises between an incompatibility between `IncremetnalClassifier` and the `feature_extraction` functions from pytorch. Because of this, I changed the mask function in the `IncrementalClassifier` class to be an in_place operation. This solves the first problem.

A second problem was a `KeyError` problem with `optimizer.state_dict()`. Because of this, I simplified the RAR implementation. This can help with the performance but sometimes take more time to run.

### CODE REVIEW

**Comment by AntonioCarta** [comment]
Is the in-place operation compatible with backpropagation? Can you use the [not in-place](https://pytorch.org/docs/stable/generated/torch.Tensor.masked_fill_.html#torch.Tensor.masked_fill_) version of the operation instead?

**Comment by JuliousHurtado** [comment]
I don't know if it is compatible con back-propagation, I don't see why not.
But yes, I change it. I run some small experiments with both versions and I don't see any difference in the performance.

Something that Albin suggest was to change the RAR plugins to something similar to ICarl, and having the feature extractor and classifier independently. This would help us remove the `name_ext_layer` parameters, since we will have a feature extractor that can return the vector. Don't know what you think?
